xtrack/tracker.py

```diff
@@ -1363,6 +1368,12 @@ def _track_no_collective(
         freeze_longitudinal=False,
         time=False
     ):
+        if type(particles) != xp.Particles:
```

Comment:
Let's add a comment specifying that the Particles class becomes part of the config so that the kernel is recompiled and store if a new Particles class is given.